homework02/rudolf/string.cpp

```diff
@@ -0,0 +1,73 @@
+#include "string.h"
+
+String::String():
+		mString(nullptr),
+		mSize(0),
+		mCapacity(0)
+{};
+String::String(const char* chars,std::size_t capacity):
+		mString(new char[capacity>strlen(chars)?capacity:strlen(chars)]),
+		mSize(strlen(chars)),
+		mCapacity(capacity>strlen(chars)?capacity:strlen(chars))
+{
+	std::copy_n(chars,mSize,mString.get());
+};
+String String::operator=(String str)
+{
+	std::swap(mString,str.mString);
+	mSize =str.size();
+	mCapacity =str.capacity();
+	return *this;
+}
+String& String::operator+(const String& str)
+{	
+	String str2(c_str(),size()+str.size());
+	for(std::size_t i=0;i<str.mSize; i++) 
+	{		
+		str2.push_back(str.mString[i]);
+	}
+	*this=str2;
+	return *this;
+}
```

Comment:
What do you expect this code to do:

```
String str = "foo";
String str2 = str + "bar";
```

I think that with your implementation, str would have been modified, which is quite surprising